### ISSUE TITLE
[SPRINT] Forecasting constraints to gravity with the MG emulator

### ISSUE THREAD

**Issue by gvalogia** [LSSTDESC/SprintWeek2020#65]
# Forecasting constraints to gravity with the MG emulator

The goal of this hack is to perform Fisher forecasting of constraints to the f(R) Hu-Sawicki MG model by the Rubin Observatory, utilizing the matter power spectrum emulator mgemu.

Contacts: @gvalogia 
Day/Time: Flexible throughout the week, Asynchronous & Synchronous EST
Main communication channel: desc-tjp-beyondwcdm and/or zoom as needed.
GitHub repo: [This link](https://github.com/LSSTDESC/mgemu/tree/Spring2020)

## Goals and deliverable
The main goals to achieve for this week:

- Predict 3D galaxy power spectrum in MG for a Rubin Observatory-like galaxy sample

- Extract covariance matrix for the fiducial cosmology, using existing large set of COLA simulations.

- Perform tomographic analysis for a realistic range of z's and produce Fisher constraints to f(R) model.

Stretch and follow-up goals are:

- Repeat the same analysis above for angular galaxy power spectrum.

- Repeat the same analysis above for lensing power spectrum and produce combined constraints.

- Repeat the same analysis above for DESI/Euclid-like 3D redshift space power spectrum and produce joint constraints.

## Resources and skills needed
Basic coding skills (python and maybe C) and interest in learning about cosmological tests of gravity. Familiarity with Fisher forecasting for surveys of the LSS.

## Detailed description
This objective is a direct follow-up of our recent paper on a [matter power spectrum emulator for f(R) gravity](https://arxiv.org/abs/2010.00596v1).
The emulator is available [in this github repository](https://github.com/LSSTDESC/mgemu)
Details about these tasks can be found at the corresponding [project page](https://confluence.slac.stanford.edu/display/LSSTDESC/Matter+Power+Spectrum+Emulator+for+f%28R%29+Modified+Gravity+Cosmologies).

**Comment by c-d-leonard** [comment]
Hey @gvalogia I'm interested to help with this!

**Comment by gvalogia** [comment]
> Hey @gvalogia I'm interested to help with this!

Awesome, I'll get in touch with you!

**Comment by nesar** [comment]
I wrote a small script to combine CCL's wCDM P(k) and MG emulator's power spectrum enhancement. Pushed a notebook [here](https://github.com/LSSTDESC/mgemu/blob/Spring2020/notebooks/pmg.ipynb). The relevant function is `pmg(Om, h, ns, s8, fR0, n, z)`

The enhancement factor is multiplied by Cosmic Emu's p(k) predictions. Some more validation needs to be done, but I have the CCL parameters set to that of COLA sims as far as I know. Here is a plot of nonlinear DM power spectra for LCDM and MG.  
![CCLemu](https://user-images.githubusercontent.com/7386526/100681249-e0f0e180-3338-11eb-845d-289ad58e64a4.png)

And this is the output from `pmg(Om, h, ns, s8, fR0, n, z)` for varying fR0 parameter. 
![pmg](https://user-images.githubusercontent.com/7386526/100681259-e3ebd200-3338-11eb-9491-3cac89ef4deb.png)

**Comment by gvalogia** [comment]
Awesome, thanks @nesar . In the meantime, I wrote a script that first evaluates the galaxy power spectra from the simulated COLA matter P(k)s, and then produces the corresponding covariance matrices. I pushed the python code and the covariances for 10 z bins (Y10 sample) and 5 z bins (Y1 sample) [here](https://github.com/LSSTDESC/mgemu/tree/Spring2020/notebooks) 
As for what the correlation matrix for the MG galaxy P(k) looks like, I added a sample plot for z=0.2 in [slide 6 here](https://docs.google.com/presentation/d/1OrI5I6it9RQgXreGbqKw-Vi5d9X0YLsKCRlnQLjtTrM/edit#slide=id.ga57b687e90_6_0)

**Comment by nesar** [comment]
This is great! so for forecasting all we need is a wrapper around P_MG(k, z) to get P_gg(k). Looks like you already have that implemented while computing the fiducial values and covariance matrix above.

**Comment by gvalogia** [comment]
Yes indeed, I can easily do that combing the script I wrote yesterday, that calculates galaxy bias and shot noise, with your function pmg. In fact, I'm thinking that this is a good thing to work on today

**Comment by gvalogia** [comment]
I just pushed an updated version of the notebook Galaxy_pmg, that evaluated the bias and shot noise factor for an LSST-like sample, and eventually the power spectrum Pgg. The notebook also loads the pre-calculated covariance matrices from the 2000 sims of the fiducial cosmology.

**Comment by gvalogia** [comment]
We have the first Fisher plots, for the Y10 sample!
![image](https://user-images.githubusercontent.com/13497222/100935347-fb07fc80-34bd-11eb-9b8a-24b54b4b10ec.png)
